chain/src/tests/util.rs

```diff
@@ -196,17 +197,54 @@ pub struct MockChain<'a> {
     blocks: Vec<BlockView>,
     parent: HeaderView,
     consensus: &'a Consensus,
+    mmr_store: MemStore<HeaderDigest>,
 }
 
 impl<'a> MockChain<'a> {
-    pub fn new(parent: HeaderView, consensus: &'a Consensus) -> Self {
+    pub fn new(
+        parent: HeaderView,
+        consensus: &'a Consensus,
+        mmr_store: MemStore<HeaderDigest>,
```

Comment:
Please remove `mmr_store` parameter and setup a default `MemStore` in the constructor since there are no `MemStore` customization requirements for current unit tests. Keep the constructor simple to make it easy to write test code.